### PR TITLE
Refactor synced GET function

### LINKED ISSUE #1133: Argument 'maxAge' must be false or a number
Whenever I minify the following example, remotestorage.js throws an error:

```js
import RemoteStorage from 'remotestoragejs'
const remoteStorage = new RemoteStorage({logging: true})
remoteStorage.access.claim('foo', 'rw')

const client = remoteStorage.scope('/foo/')
client.getListing('', false).then(listing => console.log(listing))
// error: "Argument 'maxAge' must be false or a number"
```

Note, this only happens after building the code. If you don't include remotestorage in the bundle but use a global via cdn or so, there is no error. I am not connecting to any remote storage, e.g. only local.

Could it be how the code is minified? You can test by putting above example inside a `create-react-app` (webpack) or `npx bili example.js --format umd` (rollup).

May be related to something here https://github.com/remotestorage/remotestorage.js/blob/master/src/syncedgetputdelete.js#L25

### PR DESCRIPTION
**Pull request by skddc**
Simplify the code. Aside from being more readable, this fixes a problem with the UglifyJS compressor destroying the code by restructuring it wrong.

closes #1133

### CODE REVIEW

File: src/syncedgetputdelete.js [hunk 1]
```diff
@@ -5,29 +5,27 @@ function shareFirst(path) {
...
 }
 
-function maxAgeInvalid(maxAge) {
-  return maxAge !== false && typeof(maxAge) !== 'number';
+function defaultMaxAge(context) {
+  if ((typeof context.remote === 'object') &&
+      context.remote.connected && context.remote.online) {
+    return 2 * context.getSyncInterval();
+  } else {
+    log('Not setting default maxAge, because remote is offline or not connected');
```
[the remark this answers is not in the document]
Comment (skddc) [reply src/syncedgetputdelete.js]: The function is only used one time in the code below. It's never used for not setting the default max age. So I think it's OK in this case.

**Comment by galfert** [comment]
LGTM 👍

**Comment by silverbucket** [comment]
👍

**Comment by oskarrough** [comment]
Thanks again for this. Any chance of a new release with it?

**Comment by skddc** [comment]
Yup, sometime this week!

**Pull request merged**